engine/lib/luajit-ffi-gen/README.md

```diff
@@ -150,11 +165,92 @@ For the variants without values starting index can be set, otherwise it starts f
 
 If `repr` parameter is set then `#[repr(...)]` attribute will be added with the specified type, otherwise type will be deducted from the maximal variant value: u8, u16, u32 or u64.
 
+## Joining enum and impl blocks
+
+If it's required to expose both `enum` and its `impl` block then `with_impl` attribute should be used:
```

Comment:
Just wondering, could we detect this case automatically? For example, we could perhaps do two passes when scanning the Rust AST:

* 1st pass: Add the names of all enums and impls to two hash maps, where the key is the enum / impl name.

* 2nd pass: Generate the FFI code. However, instead, when we generate the enum, we check whether an impl exists with the same name from the hash map generated in the 1st pass to decide whether to generate the impl functions. We could also skip impls if an enum exists with the same name too, if someone put an impl before the enum.

Reply:
Not sure what do you mean here.
Proc macro runs only once during compilation for each attribute entry. So it's not possible to pass a hash map from one attribute to another. That's why I'm using intermediate files in the target folder.

Reply:
Good point. When I wrote this comment, I for some reason assumed that proc macros work like an LLVM pass, where you run code against the whole AST. You can disregard what I said here :D